### DISCUSSION TITLE
externalip and extrernalhosts gossip messages

### DISCUSSION THREAD

**Discussion by johnheenan** [lightningnetwork/lnd#9073] [Q&A] [answered]
I understand `--externalip` on the command line is used to advertise channels can be opened.

I understand `externalhosts` in lnd.conf is used to advertise to peers a new IP address for channels

Is there any difference between how these messages are propagated and interpreted in p2p gossip?

For example, I would expect a peer to further pass on `externalip` IP gossip but not pass on `extrnalhosts` IP gossip. Is this correct?

Suppose an IP addess has changed. Is `externalip` with the server LN name gossip on a restart sufficent to reauthenticate with public keys so channels will be reopened in a timely manner?

**Comment by guggero** [comment, marked as answer]
> Is there any difference between how these messages are propagated and interpreted in p2p gossip?

No. The result is just a list of IP addresses that's being gossiped. The list is the combination of the `externalip` static IP addresses and the periodically _resolved_ IP addresses from the `externalhosts` configs. As mentioned previously, no domain names are propagated, only IP addresses. So another node cannot distinguish between static IP addresses and those resolved from the host name given in `externalhosts`.

> restart sufficent to reauthenticate

A restart is not needed. The host names in `externalhosts` are resolved every 5 minutes. But it can take some time for the new gossip message to reach all nodes in the network, which is the real bottleneck most likely (e.g. if you're waiting for a public network explorer to show the change, that can take up to 24 hours). Your node's peer should get the update immediately though, assuming it is connected. And an IP change normally doesn't cause an interruption in the p2p connection. So they should get the new IP, store it and then attempt to connect on that the next time a connection attempt is being made.

**Reply by johnheenan** [reply]
Thanks for your reponse. I have a new upcoming question and I will be quoting from your answer here.
